File: UsedProductExchangeSolution/UsedProductExchange.Core.Entities/Product.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace UsedProductExchange.Core.Entities
{
    public class Product
    {
        public int ProductId { get; set; }
        public int UserId { get; set; }
        public User User { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string PictureUrl { get; set; }
        public double CurrentPrice { get; set; }
        public DateTime Expiration { get; set; }
        public int CategoryId { get; set; }
        public Category Category { get; set; }
        public bool IsSold { get; set; }
        
        public virtual ICollection<Bid> Bids { get; set; }
    }
}

File: UsedProductsExchangeFrontend/src/app/_models/product.ts
import { Bid } from '../_models/Bid';

export class Product {
  productId?: number;
  userId: number;
  name: string;
  description: string;
  pictureUrl: string;
  currentPrice: number;
  expiration: any;
  categoryId: number;
  bids: Bid[];
}
